### PR TITLE
[WIP] Add connector documentation

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: velox/docs/develop/connectors.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+==========
+Connectors
+==========
+
+Connectors allow Velox to read and write data to and from external sources.
+This concept is similar to `Presto Connectors <https://prestodb.io/docs/current/develop/connectors.html>`_.
+The :ref:`TableScanNode<TableScanNode>` operator reads data from a connector.
```
Comment (mbasmanova) [inline velox/docs/develop/connectors.rst:7]: Maybe mention that TableWrite operator writes data through a connector.

File: velox/docs/develop/connectors.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     - Provides methods to consume and process a split. A DataSource can optionally consume a
+       dynamic filter during execution to prune some rows from the output vector.
+   * - DataSink
+     - Provides methods to write a Velox vector externally.
+   * - ConnectorQueryCtx
+     - Interface to specify configuration to a DataSink or a DataSource.
+   * - Connector
+     - This is the main interface that allows creating instances of a DataSource or a DataSink.
+   * - Connector Factory
+     - This interface enables creating instances of connectors using the specified name.
```
Comment (mbasmanova) [inline velox/docs/develop/connectors.rst:31]: It would be nice to clarify the difference between connector name and ID.

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Deepak, thank you for starting on this.

File: velox/docs/develop/connectors.rst (hunk 2 above)
Comment (majetideepak) [reply velox/docs/develop/connectors.rst]: done

**Comment by majetideepak** [comment]
All CI tests are failing due to a git branch issue. Closing this PR.

**Pull request closed without merging**